security_group_inputs.tf

```
@@ -0,0 +1,88 @@
+# security_group_inputs Version: 1
+#
+
+variable "create_security_group" {
+  type        = bool
+  default     = true
+  description = "Set `true` to create and configure a new security group. If false, `associated_security_group_ids` must be provided."
+}
+
+variable "associated_security_group_ids" {
+  type        = list(string)
+  default     = []
+  description = <<-EOT
+    A list of IDs of Security Groups to associate the created resource with, in addition to the created security group.
+    These security groups will not be modified and, if `create_security_group` is `false`, must have rules providing the desired access.
+    EOT
+}
+
+variable "allowed_security_group_ids" {
+  type        = list(string)
+  default     = []
+  description = <<-EOT
+    A list of IDs of Security Groups to allow access to the security group created by this module.
+    EOT
+}
+
+locals {
+  allowed_security_group_ids = concat(var.security_groups, var.allowed_security_group_ids)
+}
+
+variable "security_group_name" {
+  type        = list(string)
+  default     = []
+  description = <<-EOT
+    The name to assign to the created security group. Must be unique within the VPC.
+    If not provided, will be derived from the `null-label.context` passed in.
+    If `create_before_destroy` is true, will be used as a name prefix.
+    EOT
+}
+
+variable "security_group_description" {
+  type        = string
+  default     = "EFS Security Group"
+  description = <<-EOT
+    The description to assign to the created Security Group.
+    Warning: Changing the description causes the security group to be replaced.
+    EOT
+}
+
+variable "security_group_create_before_destroy" {
+  type = bool
+
+  default     = false
+  description = <<-EOT
+    Set `true` to enable Terraform `create_before_destroy` behavior on the created security group.
+    We recommend setting this `true` on new security groups, but default it to `false` because `true`
+    will cause existing security groups to be replaced, possibly requiring the resource to be deleted and recreated.
+    Note that changing this value will always cause the security group to be replaced.
```

The old security group [had create_before_destory set
](https://github.com/cloudposse/terraform-aws-efs/blob/7d85eb917f1381bc329cee2433acfbbfbdf695a8/main.tf#L60-L62) so keep it set.
```suggestion
  default     = true
  description = <<-EOT
    Set `true` to enable Terraform `create_before_destroy` behavior on the created security group.
    Note that changing this value will always cause the security group to be replaced.
```